daemon/cmd/hubble.go

```diff
@@ -172,7 +173,19 @@ func (d *Daemon) launchHubble() {
 		}
 	}
 
-	d.hubbleObserver, err = observer.NewLocalServer(payloadParser, logger,
+	namespaceManager := observer.NewNamespaceManager()
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		namespaceManager.Run(d.ctx)
+		wg.Done()
+	}()
+	defer wg.Wait()
```

Comment:
Taking another look, before this patch, `launchHubble` is [executed into its own goroutine](https://github.com/cilium/cilium/blob/3e805306fc4125fca3398397acdf11a13f1a9600/daemon/cmd/daemon_main.go#L1871). In the happy path the "launch" goroutine setup a goroutine for the "local server" and a corresponding cleanup goroutine, another goroutine for the "remote server" and a corresponding cleanup goroutine, and then returns. So we start the "launch" goroutine, itself starting 4 other goroutines, and then the "launch" goroutine terminates.

My suggestion then doesn't work because when launchHubble returns then the context is cancelled, and the `namespaceManager` shutdown.

In the current version of the patch, the "launch" goroutine will be stuck on the deferred `wg.Wait()` until `d.ctx` is cancelled, cancelling the local `ctx` and shutting down the `namespaceManager` (which will eventually call `wg.Done()` and allow the "launch" goroutine to return from `launchHubble` and terminate).

If correct, although I agree with you wrt best practice on principle, practically I don't see the point of keeping the waitGroup. Technically it only make it so the "launch" goroutine stick around, which have any purpose anymore. As I see it it is one more goroutine that is sleeping or that we might leak.